controllers/topology/internal/mergepatch/mergepatch.go

```diff
@@ -98,28 +109,41 @@ func NewHelper(original, modified client.Object, c client.Client, opts ...Helper
 }
 
 // filterPatch removes from the patch diffs not in the allowed paths;
-// it also return a flag indicating if the resulting patch has spec changes or not.
-func filterPatch(patch []byte, allowedPaths, ignorePaths []contract.Path) ([]byte, bool, error) {
-	// converts the patch into a Map
-	patchMap := make(map[string]interface{})
+// it also returns a flag indicating if the resulting patch has spec changes or not.
+func filterPatch(simplePatch, twoWayPatch []byte, allowedPaths, ignorePaths []contract.Path, authoritativePaths []contract.Path) ([]byte, bool, error) {
+	twoWayPatchMap := make(map[string]interface{})
+	if err := json.Unmarshal(twoWayPatch, &twoWayPatchMap); err != nil {
+		return nil, false, errors.Wrap(err, "failed to unmarshal two way merge patch")
+	}
+
+	// Enforce changes from simple patch when required.
+	// This will override instance specific fields for a subset of fields,
+	// e.g. machine template metadata changes should be reflected into generated objects without
+	// accounting for instance specific changes like we do for other maps into spec.
+	if len(authoritativePaths) > 0 {
+		simplePatchMap := make(map[string]interface{})
+		if err := json.Unmarshal(simplePatch, &simplePatchMap); err != nil {
+			return nil, false, errors.Wrap(err, "failed to unmarshal simple merge patch")
+		}
 
-	if err := json.Unmarshal(patch, &patchMap); err != nil {
-		return nil, false, errors.Wrap(err, "failed to unmarshal merge patch")
+		for _, path := range authoritativePaths {
+			enforcePath(simplePatchMap, twoWayPatchMap, path)
+		}
 	}
 
-	// Removes from diffs everything not in the allowed paths.
-	filterPatchMap(patchMap, allowedPaths)
+	// Removes from diffs everything not in allowed paths.
+	filterPatchMap(twoWayPatchMap, allowedPaths)
```

Comment:
Check to see if the enforcedPatch is in the allowedPaths to avoid unexpected effects for users?

Reply:
I have added a comment to top of NewHelper give that allowedPaths are hard coded and applies to the entire func.